src/server/document-controller.js

```diff
@@ -129,6 +130,26 @@ class DocumentController {
       return res.send({ documentRevisions });
     });
 
+    router.patch('/api/v1/docs/:key/archive', needsPermission(permissions.MANAGE_ARCHIVED_DOCS), async (req, res) => {
```

Comment:
This is not what I meant by patch, more like this:
https://restcookbook.com/HTTP%20Methods/patch/
I think it does exactly what you want it do do and it can be used for other updates as well. Again we don't need to go this route, it's just the more restful way.

Reply:
Maybe a better link:
https://stackoverflow.com/questions/28459418/use-of-put-vs-patch-methods-in-rest-api-real-life-scenarios

Reply:
I don't get it, based on the links you did mean PATCH as the HTTP verb, or?

Reply:
Yes, but the implementation here is not according to "spec":
 you could have 
api/v1/docs/:key
verb: PATCH
body:
{
   archived: true/false
}

Then for future such updates we could avoid to make new endpoints. The caveats are that these are not idempotent operations and we could introduce some race conditions. 

The reason why I suggested this was that you wanted yesterday to avoid doing a PUT which would need to send the whole object.

Reply:
Ah ok, thanks for explaining what you meant, that helps 😬 
I wrote here https://github.com/educandu/educandu/pull/8#discussion_r745358276 why I preferred to go with 2 endpoints.